index.html

```diff
@@ -4967,33 +4938,33 @@ <h3 id="looping-AudioBufferSourceNode">
             Looping
           </h3>
           <p>
-            If the <code>loop</code> attribute is true when
-            <code>start()</code> is called, then playback will continue
+            If the <code>loop</code> attribute is true when <a href=
+            "#widl-AudioBufferSourceNode-start-void-double-when-double-offset-double-duration">
+            <code>start()</code></a> is called, then playback will continue
             indefinitely until <code>stop()</code> is called and the stop time
             is reached. We'll call this "loop" mode. Playback always starts at
             the point in the buffer indicated by the <code>offset</code>
-            argument of <code>start()</code>, and in <em>loop</em> mode will
-            continue playing until it reaches the <em>actualLoopEnd</em>
-            position in the buffer (or the end of the buffer), at which point
-            it will wrap back around to the <em>actualLoopStart</em> position
-            in the buffer, and continue playing according to this pattern.
+            argument of <a href=
+            "#widl-AudioBufferSourceNode-start-void-double-when-double-offset-double-duration">
+            <code>start()</code></a>, and in <em>loop</em> mode will continue
+            playing until it reaches the <em>loopEnd</em> position in the
+            buffer (or the end of the buffer), at which point it will wrap back
+            around to the <em>loopStart</em> position in the buffer, and
+            continue playing according to this pattern.
           </p>
           <p>
-            In <em>loop</em> mode then the <em>actual</em> loop points are
-            calculated as follows from the <code>loopStart</code> and
-            <code>loopEnd</code> attributes:
+            The effective loop start and end points are required to lie within
+            the range of zero and the buffer duration, as specified in the
+            algorithm below. <code>loopEnd</code> is further constrained to be
+            at or after <code>loopStart</code>. If any of these constraints are
+            violated, the loop is considered to include the entire buffer
+            contents.
+          </p>
+          <p>
+            Loop-related properties may be varied during playback of the
+            buffer. The exact results are defined by the playback algorithm
+            which follows.
```

Comment:
Sorry about being a bit unclear. I meant something along the line of:

We have two ways of speccing the behaviour of the `AudioBufferSourceNode`, using text and using an algorithm. We should make the text non-normative, and the algorithm normative, so that we have a single authoritative resource to know if a particular behaviour is per spec or not.

The english text and the diagrams are useful to understand the general behaviour of things, and why it is specced like it is, but I don't think it should be normative, since it's open for interpretation and not as precise as the algorithm.